src/js/_enqueues/admin/site-icon.js

```diff
@@ -0,0 +1,152 @@
+						control: {
+							params: {
+								width: $el.data( 'size' ),
+								height: $el.data( 'size' )
+							}
+						},
+						imgSelectOptions: calculateImageSelectOptions
+					})
+				]
+			});
+
+			frame.on( 'cropped', function( attachment) {
+				$( '#site_icon_hidden_field' ).val(attachment.id);
+				switchToUpdate(attachment.url);
+				frame.close();
+				// Startover with a frame that is so fresh and so clean clean.
+				frame = null;
+			});
+
+			// When an image is selected, run a callback.
+			frame.on( 'select', function() {
+				// Grab the selected attachment.
+				var attachment = frame.state().get('selection').first();
+
+				if ( attachment.attributes.height === $el.data('size') && $el.data('size') === attachment.attributes.width ) {
+					// set the value of the hidden input to the attachment id
+					$( '#site_icon_hidden_field').val(attachment.id);
+					switchToUpdate(attachment.attributes.url);
+					// close the model
+					frame.close();
+				} else {
+					frame.setState( 'cropper' );
+				}
+			});
+
+			frame.open();
+		});
+	});
+
+	function switchToUpdate( url ){
+		// set site-icon-img src to the url and remove the hidden class
+		$( '#site-icon-img' ).attr( 'src', url );
+		$( '#site-icon-img' ).removeClass( 'hidden' );
+		// remove hidden class from remove
+		$( '#js-remove-site-icon' ).removeClass( 'hidden' );
+		// tmp store the classes of the #choose-from-library-link button
+		// if the button is not in the update state, swap the classes
+		if( $( '#choose-from-library-link' ).attr('data-state') !== '1' ){
+			var classes = $( '#choose-from-library-link' ).attr( 'class' );
+			$( '#choose-from-library-link' ).attr('class', $( '#choose-from-library-link' ).attr('data-alt-classes') );
+			$( '#choose-from-library-link' ).attr('data-alt-classes', classes );
+			$( '#choose-from-library-link' ).attr('data-state', '1');
+		}
+
+		// swap the text of the button
+		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr('data-update-text') );
+	}
+
+	$( '#js-remove-site-icon' ).on( 'click', function( event ) {
+		// set the value of the hidden input to blank
+		$( '#site_icon_hidden_field').val('false');
+		//  toggle hidden class on site-icon-img
+		$( '#site-icon-img' ).toggleClass( 'hidden' );
+		// toggle hidden class on remove button
+		$( this ).toggleClass( 'hidden' );
+
+		// tmp store the classes of the #choose-from-library-link button
+		var classes = $( '#choose-from-library-link' ).attr('class');
+		$( '#choose-from-library-link' ).attr('class', $( '#choose-from-library-link' ).attr('data-alt-classes') );
+		$( '#choose-from-library-link' ).attr('data-alt-classes', classes );
+
+		// swap the text of the button
+		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr('data-choose-text') );
+
+		$( '#choose-from-library-link' ).attr('data-state', '');
+
+		// prevent the default action
+		event.preventDefault();
```

Comment:
By making sure the button has a `type="button"` attribute, there's no default action to prevent and this can be removed.